src/backend/commands/tablecmds.c

```diff
@@ -17666,6 +17666,12 @@ static Datum transformExecOnClause(List	*on_clause, int *preferred_segment_num)
 		}
 		else if (strcmp(defel->defname, "master") == 0)
 		{
+			if(iswritable){
+			  ereport(ERROR,
+	 	      (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+			    errmsg("the ON master syntax for writable external tables is deprecated"),
```

Comment:
would it be possible to add a hint saying what is the allowed option (ON number)?